packages/boilerplate/src/index.js

```diff
@@ -1,16 +1,37 @@
 import defaults from './lib/defaults';
 import factory from './lib/factory';
 
+/*
+ * Converts a passed selector which can be of varying types into an array of DOM Objects
+ *
+ * @param selector, Can be a string, Array of DOM nodes, a NodeList or a single DOM element.
+ */
+export const getSelection = (selector) => {
+    let nodes = [];
+
+    if(typeof selector === "string") {
+        nodes = [].slice.call(document.querySelectorAll(selector));
+    } else if (selector instanceof Array) {
+        nodes = selector;
+    } else if (Object.prototype.isPrototypeOf.call(NodeList.prototype, selector)) {
+        nodes = [].slice.call(selector);
+    } else if (selector instanceof HTMLElement) {
+        nodes.push(selector)
+    }
+
+    return nodes;
```

Comment:
Nitpick, but could we do without the nodes array, and the else statements and just if and return?

Something like
if (typeof selector === "string") return [].slice.call(document.querySelectorAll(selector));
if (selector instanceof Array) return selector;
etc

I won't be too disappointed if you disagree, I don't think the removing the additional variable or logic would have any measurable performance improvements.

Reply:
Sure thing - I've pushed a change to the function.  Is that the type of thing you were thinking?  Definitely neatens it. 

Thanks!
Sarah